src/app/Library/CrudPanel/Traits/FieldsProtectedMethods.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+namespace Backpack\CRUD\app\Library\CrudPanel\Traits;
+
+trait FieldsProtectedMethods
+{
+    /**
+     * The only REALLY MANDATORY attribute when defining a field is the 'name'.
+     * Everything else Backpack can probably guess. This method makes sure  the
+     * field definition array is complete, by guessing missing attributes.
+     *
+     * @param  string|array $field  The definition of a field (string or array).
+     * @return array                The correct definition of that field.
+     */
+    protected function makeSureFieldHasNecessaryAttributes($field)
+    {
+        $field = $this->makeSureFieldHasName($field);
+        $field = $this->makeSureFieldHasModel($field);
+        $field = $this->makeSureFieldHasLabel($field);
+        $field = $this->makeSureFieldHasType($field);
+        $field = $this->makeSureFieldHasRelationshipData($field);
+
+        return $field;
+    }
+
+    /**
+     * If the field_definition_array array is a string, it means the programmer was lazy
+     * and has only passed the name of the field. Turn that into a proper array.
+     *
+     * @param  string|array $field The field definition array (or string).
+     * @return array
+     */
+    protected function makeSureFieldHasName($field)
+    {
+        if (is_string($field)) {
+            return ['name' => $field];
+        }
+
+        if (is_array($field) && ! isset($field['name'])) {
+            abort(500, 'All fields must have their name defined');
+        }
+
+        return $field;
+    }
+
+    protected function makeSureFieldHasRelationshipData($field)
+    {
+        // only do this if "entity" is defined on the field
+        if (! isset($field['entity'])) {
```

Comment:
I think we should invert this check to see first if field type is relationship. If it is relationship developer could have give the name `category`, `category_id` or `categories` in case it's a multiple relation. 

If he provide `category` as a name, and exists a category relation defined we get the relation data from there and swap the field name accordingly to `category_id` if that is the key in relation.

If you provide `category_id`, it will still work, in relationship trait we get model public methods that expose the return type (PHP 7++) **We use this technique to avoid running all methods on model and just the ones that expose the relations.**
```php
function category() : BelongsTo
{
   return $this->belongsTo('Backpack\NewsCRUD\app\Models\Category');
}
```

ping @tabacitu